core/print_api.php

```diff
@@ -1894,16 +1894,19 @@ function print_bug_attachments_list( $p_bug_id, $p_security_token ) {
  * a valid security token, previously generated by form_security_token().
  * Use this to avoid performance issues when loading pages having many calls to
  * this function, such as print_bug_attachments_list().
- * @param array $p_attachment An attachment array from within the array returned by the file_get_visible_attachments() function.
+ * @param array $p_attachment An attachment array from within the array returned
+ *                            by the file_get_visible_attachments() function.
  * @param string $p_security_token The security token to use for deleting attachments.
  * @return void
  */
 function print_bug_attachment( array $p_attachment, $p_security_token ) {
-	$t_show_attachment_preview = $p_attachment['preview'] && $p_attachment['exists'] && ( $p_attachment['type'] == 'text' || $p_attachment['type'] == 'image' );
+	$t_show_attachment_preview = $p_attachment['preview']
+		&& $p_attachment['exists']
+		&& ( $p_attachment['type'] == 'text' || $p_attachment['type'] == 'image' );
 	if( $t_show_attachment_preview ) {
 		$t_collapse_id = 'attachment_preview_' . $p_attachment['id'];
 		global $g_collapse_cache_token;
-		$g_collapse_cache_token[$t_collapse_id] = false;
+		$g_collapse_cache_token[$t_collapse_id] = $p_attachment['type'] == 'image';
```

Comment:
> the idea of 'preview' flag is to determine whether we should auto-preview 

Actually my understanding of the preview flag (I guess, based on how it's been implemented until now) was that it indicates the *ability* to view the attachment inline, as opposed to automatically displaying it.

While I like to view images directly, I would prefer not to expand text files inline by default, as it can unnecessarily eat up a lot of vertical space. I don't think this specific handling can be achieved if we move the logic to file API.